src/components/RichEditor/LanguagePlugin.js

```diff
@@ -0,0 +1,132 @@
+import tinymce from 'tinymce/tinymce.min';
+import { BROWSER_DEFAULT, languages } from '../../data/constants/formOptions';
+import updateFormLanguage from '../../data/actions/formOptions';
+import { getLanguageforEditor } from '../../data/reducers/formOptions';
+import store from '../../data/store';
+
+tinymce.PluginManager.add('language', (editor) => {
```

Comment:
I think it would be helpful to leave comments throughout this file explaining what is happening at different locations. Places I think that could use commenting:

1. All of the `selectedNode.nodeName` if conditions and the differences in the results there.
2. the `resetLanguage` function

If there are any other spots you think could be unclear, please also comment those.